runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -2929,6 +2980,26 @@ void set_hugepage_info(void)
     hugepage_size = gethugepagesize();
   }
 
+  //
+  // With HUGETLB_NO_RESERVE=yes, we want to report out-of-memory in
+  // response to SIGBUS signals resulting from (as best we can tell)
+  // being unable to acquire hugepages at fault-in time.
+  //
+  {
+    char const *ev;
+
+    if ((ev = getenv("HUGETLB_NO_RESERVE")) != NULL
+        && strcasecmp(ev, "yes") == 0) {
+      struct sigaction act;
+
+      act.sa_flags = SA_RESTART | SA_SIGINFO;
+      act.sa_sigaction = SIGBUS_handler;
+      if (sigaction(SIGBUS, &act, &previous_SIGBUS_sigact) != 0) {
```

Comment:
Since `act` is an automatic variable, the signal mask might contain random bits.  You'll need something like this at the same time that you're initializing the other fields.
```c
sigemptyset(&act.sa_mask);
```

Reply:
Ugh, how did I miss that?  Will fix ...